ddtrace/contrib/futures/threading.py

```diff
@@ -17,9 +20,10 @@ def _wrap_submit(func, args, kwargs):
     #
     #      The resolution is to not create/propagate a new context if one does not exist, but let the
     #      future's thread create the context instead.
-    current_ctx = None
+    # DEV: Be sure to propagate a Context and not a Span since we are crossing thread boundaries
+    current_ctx: Optional[Context] = None
```

Comment:
```suggestion
    current_ctx: Optional[Context] = ddtrace.tracer.current_trace_context()
```

Reply:
right! I was staring at the code too long, I thought that the dev comment above was still true.